# Larissa Kidd 9/12/2023
## Instragram Likes Report
 I hypothesize that "pubity" and "memezar" are in my top 20 accounts liked  because I have followed them for a long time and I like pubity's news reports and I enjoy memes

In [26]:
import pandas as pd
import json

In [27]:
with open('/Users/larissakidd/Downloads/Fourth Year/Fall 23/Data in EMAT/instagram/likes/liked_posts.json') as j:
    my_data = json.load(j)
my_data

{'likes_media_likes': [{'title': 'pubity',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwpdJOQMAaF/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693581754}]},
  {'title': 'kentsportswear',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwpZ71suXc1/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693581074}]},
  {'title': 'larochesoccer',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwpxFoCLUkJ/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693581036}]},
  {'title': 'memezar',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwnxypUI467/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693515116}]},
  {'title': 'hannah.lohmueller',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwnUu6zMiqQ/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693515083}]},
  {'title': '_abbydev_',
   'string_list_data': [{'href': 'https://www.instagram.com/p/Cwm_eNTLXt2/',
     'value': 'ð\x9f\x91\x8d',
    

In [28]:
my_df = pd.DataFrame(my_data['likes_media_likes'])

In [29]:
my_data['likes_media_likes'][0]['string_list_data'][0]['timestamp']

1693581754

In [30]:
time = [x['string_list_data'][0]['timestamp'] for x in my_data['likes_media_likes']] 
time

[1693581754,
 1693581074,
 1693581036,
 1693515116,
 1693515083,
 1693493787,
 1693493777,
 1693447022,
 1693428135,
 1693417571,
 1693417490,
 1693416823,
 1693416446,
 1693403266,
 1693367393,
 1693332733,
 1693267562,
 1693267522,
 1693267465,
 1693245279,
 1693156403,
 1693005560,
 1692970162,
 1692828581,
 1692814078,
 1692758190,
 1692758007,
 1692757962,
 1692757902,
 1692757885,
 1692757854,
 1692736697,
 1692736688,
 1692736640,
 1692672707,
 1692647140,
 1692646797,
 1692628164,
 1692628140,
 1692589267,
 1692484191,
 1692484171,
 1692484167,
 1692484123,
 1692484060,
 1692483826,
 1692483782,
 1692483484,
 1692457604,
 1692450922,
 1692385771,
 1692384623,
 1692379378,
 1692378555,
 1692378516,
 1692378440,
 1692378175,
 1692370300,
 1692369946,
 1692369934,
 1692367232,
 1692365641,
 1692365253,
 1692360577,
 1692360530,
 1692360368,
 1692360338,
 1692360324,
 1692360226,
 1692360201,
 1692360147,
 1692359920,
 1692325341,
 1692325333,
 1692325232,
 1692323246,
 1692322672,

In [31]:
my_df['likes_timestamps'] = time
my_df

title                                   string_list_data  \
0                 pubity  [{'href': 'https://www.instagram.com/p/CwpdJOQ...   
1         kentsportswear  [{'href': 'https://www.instagram.com/p/CwpZ71s...   
2          larochesoccer  [{'href': 'https://www.instagram.com/p/CwpxFoC...   
3                memezar  [{'href': 'https://www.instagram.com/p/CwnxypU...   
4      hannah.lohmueller  [{'href': 'https://www.instagram.com/p/CwnUu6z...   
...                  ...                                                ...   
8932          arieljean_  [{'href': 'https://www.instagram.com/p/jiLKwbq...   
8933  justgirlyythaaangs  [{'href': 'https://www.instagram.com/p/jgoK0_B...   
8934      teendefinition  [{'href': 'https://www.instagram.com/p/jY9kJuk...   
8935  justgirlyythaaangs  [{'href': 'https://www.instagram.com/p/jIE3pgh...   
8936         pv_boii2000  [{'href': 'https://www.instagram.com/p/gwrBtYy...   

      likes_timestamps  
0           1693581754  
1           1693581074  
2           1693581036  
3           1693515116  
4           1693515083  
...                ...  
8932        1390531445  
8933        1390477492  
8934        1390415033  
8935        1390262413  
8936        1390261066  

[8937 rows x 3 columns]

In [32]:
grouped = my_df.groupby('title').count()
grouped.shape

(1382, 2)

In [33]:
grouped.sort_values('likes_timestamps', ascending = False).head(20)

string_list_data  likes_timestamps
title                                                      
cohmedy                               199               199
hhigley00                             144               144
lilygcashmoneyap                      144               144
ksudeltagamma                         137               137
pubity                                120               120
zk_k18                                108               108
fab_ksu                               104               104
lilygracepittman                      100               100
xo.aubs_                               81                81
daquan                                 78                78
kellysummersphotography                74                74
barstoolkent                           64                64
mat_thew_please                        58                58
courtney_sayre                         55                55
theanairene                            53                53
tampon                                 53                53
ksudgphilanthropy                      49                49
jenicakidd                             48                48
scoobydoofruitsnacks                   48                48
shockran99                             46                46

In [34]:
grouped.loc['memezar']

string_list_data    34
likes_timestamps    34
Name: memezar, dtype: int64

## While "pubity" is in the top 5 with 120 likes,  "memezar" is well below the top 20 with only 34 likes. 
I also realized that I neglected to think about my long time best friend, my sorority, and my boyfriend which are also obviously in the top 5... whoops

## If I could reapproach..
### I would figure out the timeline of how far back this data goes 
It is hard to read the timestamps because of their format so I might have misjudged based on how far back this data is collected from